src/ProductConstructionService/ProductConstructionService.Api/Api/v2020_02_20/Controllers/SubscriptionsController.cs

```diff
@@ -396,6 +396,15 @@ public async Task<IActionResult> Create([FromBody, Required] SubscriptionData su
                 "The Maestro github application must be installed by the repository's owner and given access to the repository."
             ]));
         }
+        // We also need to make sure the source repository is registered if the subscription is source-enabled
+        if (subscription.SourceEnabled ?? false && !(await EnsureRepositoryRegistration(subscription.SourceRepository)))
+        {
+            return BadRequest(new ApiError("The request is invalid",
+            [
+                $"No Maestro GitHub application installation found for repository '{subscription.SourceRepository}'. " +
+                "The Maestro github application must be installed by the repository's owner and given access to the repository."
+            ]));
+        }
```

Comment:
I think it's better to do what we did in https://github.com/dotnet/arcade-services/commit/5e4e719b82d1b8819051b704cde17a79ad35c481 and just look up the registration during the subscription creation (and fail there if there indeed is no installation present).

Reply:
maybe we don't actually need this, I added it to my local when I was running the repro tool on my new devbox and didn't have any repos in there